sound/soc/sof/intel/hda-dsp.c

```diff
@@ -282,6 +282,50 @@ void hda_dsp_ipc_int_disable(struct snd_sof_dev *sdev)
 			HDA_DSP_REG_HIPCCTL_BUSY | HDA_DSP_REG_HIPCCTL_DONE, 0);
 }
 
+int hda_dsp_update_d0i3c(struct snd_sof_dev *sdev, bool enable)
+{
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	u8 reg;
+	int retry = 50;
+
+	reg = snd_hdac_chip_readb(bus, VS_D0I3C);
+	/* Write to D0I3C after Command-In-Progress bit is cleared */
+	while ((reg & SOF_HDA_VS_D0I3C_CIP) && --retry) {
+		usleep_range(10, 15);
+		reg = snd_hdac_chip_readb(bus, VS_D0I3C);
```

Comment:
This pattern occurs so often - when you have to perform a calculation to verify a condition and then repeat an action while the condition holds, so you end up with
```
test = calculate_test();
while (must_loop(test)) {
	action(test);
	test = calculate_test();
}
```
that maybe we have to add an exception to kernel coding style guidelines to allow
```
while (must_loop(test = calculate_test()))
	action(test);
```
but for now we have to do the former...

Reply:
So what's your suggestion here @lyakh ?

Reply:
@keyonjie write to LKML to suggest add this exception to the Coding Style :-)